File: PoolDN/PoolDN/Models/Match.swift
import Foundation

enum MatchStatus: String, Codable {
    case scheduled
    case inProgress = "in_progress"
    case pendingReview = "pending_review"
    case completed
}

struct GameResult: Codable {
    let gameOrder: Int
    let homePlayerName: String?
    let awayPlayerName: String?
    let homeScore: Int
    let awayScore: Int
}

struct ScoreSubmission: Codable {
    let homeScore: Int
    let awayScore: Int
    let games: [GameResult]?
    let submittedBy: String
    let submittedAt: Int
}

struct Match: Codable, Identifiable {
    let id: String
    let competitionId: String
    let round: Int
    let matchday: Int
    let homeTeamId: String
    let awayTeamId: String
    let homeTeamName: String
    let awayTeamName: String
    let scheduledDate: String?
    let venue: String?
    let status: MatchStatus
    let homeScore: Int
    let awayScore: Int
    let games: [GameResult]?
    let homeSubmission: ScoreSubmission?
    let awaySubmission: ScoreSubmission?
    let confirmedBy: String?
    let submittedBy: String?
    let createdAt: Int
    let updatedAt: Int
}

File: api/src/models/match.ts
import { z } from 'zod';
import { baseEntitySchema } from './base';

export const matchStatus = z.enum(['scheduled', 'in_progress', 'pending_review', 'completed']);
export type MatchStatus = z.infer<typeof matchStatus>;

export interface MatchSubmission {
  homeScore: number;
  awayScore: number;
  games?: GameResult[];
  submittedBy: string;
  submittedAt: number;
}

export const gameResultSchema = z.object({
  gameOrder: z.number(),
  homePlayerName: z.string().optional(),
  awayPlayerName: z.string().optional(),
  homeScore: z.number().default(0),
  awayScore: z.number().default(0),
});

export type GameResult = z.infer<typeof gameResultSchema>;

export const matchSchema = baseEntitySchema.extend({
  competitionId: z.string(),
  round: z.number(),
  matchday: z.number(),
  homeTeamId: z.string(),
  awayTeamId: z.string(),
  homeTeamName: z.string(),
  awayTeamName: z.string(),
  scheduledDate: z.string().optional(),
  venue: z.string().optional(),
  status: matchStatus.default('scheduled'),
  homeScore: z.number().default(0),
  awayScore: z.number().default(0),
  games: z.array(gameResultSchema).optional(),
  homeSubmission: z.any().optional(),
  awaySubmission: z.any().optional(),
  confirmedBy: z.string().optional(),
  submittedBy: z.string().optional(),
});

export type Match = z.infer<typeof matchSchema>;
